deploy/operator/setup_hive.sh

```diff
@@ -16,7 +46,57 @@ spec:
   sourceNamespace: openshift-marketplace
 EOCR
 
-wait_for_operator "hive-operator" "openshift-operators"
-wait_for_crd "clusterdeployments.hive.openshift.io"
+  wait_for_operator "hive-operator" "openshift-operators"
+  wait_for_crd "clusterdeployments.hive.openshift.io"
+
+  echo "Hive installed successfully!"
+}
+
+function from_upstream() {
+  HIVE_DIR="${HIVE_DIR:-${HOME}/go/src/github.com/openshift/hive}"
+
+  if [ ! -d "${HIVE_DIR}" ]; then
+    git clone https://github.com/openshift/hive.git "${HIVE_DIR}"
+  fi
+
+  pushd $HOME/go/src/github.com/openshift/hive
+  git fetch origin
+  git pull origin master
+
+  if [ "${DISCONNECTED}" = "true" ]; then
+    export IMG="${LOCAL_REGISTRY}/localimages/hive:latest"
+    oc image mirror \
+       -a ${AUTHFILE} \
+       ${HIVE_IMAGE} \
+       ${IMG}
+  else
+    export IMG="${HIVE_IMAGE}"
+  fi
+
+  make deploy
+  wait_for_pod "hive-operator" "hive" "control-plane=hive-operator"
+  wait_for_pod "hive-controllers" "hive" "control-plane=controller-manager"
+
+  popd
+}
+
+function install_dependencies() {
+  echo "Installing kustomize..."
+  (cd /usr/local/bin && curl -s "https://raw.githubusercontent.com/\
+kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash)
+
+  echo "Installing golang..."
+  curl -L https://storage.googleapis.com/golang/getgo/installer_linux -o /tmp/golang_installer
+  chmod u+x /tmp/golang_installer
+  /tmp/golang_installer
+  rm /tmp/golang_installer
+
+  echo "Activating go command on current shell..."
+  set +u
+  source /root/.bash_profile
+  set -u
+}
```

Comment:
I would like to avoid another place with installations. There's already `setup_env.sh` for that

Reply:
Which installs way too much other stuff that we don't need on a dev-scripts environment...
Maybe I'll make it an optional step with an env-var? Or non-optional

Reply:
I think this is something we are going to have to reconcile maybe in a later story. `setup_env.sh` seems to include CI dependencies to run on the prow k8s cluster so it is a bit different.

Reply:
We can split setup_env.sh into sections (functions). 
Eventually, I don't want us to have many installations scripts that we will need to maintain (just as an example - I just updated 5 files with `curl retry`)

Reply:
@YuviGold we can talk about splitting ``setup_env.sh``
but keep in mind that it's better to keep dependency installation closer to its usage. Changing 5 files to allow retrying is the lesser evil :)